tests/ResourceSyncingTest.php

```diff
@@ -628,9 +876,106 @@ public function getCentralModelName(): string
     public function getSyncedAttributeNames(): array
     {
         return [
+            'global_id',
             'name',
             'password',
             'email',
         ];
     }
 }
+
+// override method in ResourceUser class to return default attribute values
+class ResourceUserProvidingDefaultValues extends ResourceUser
+{
+    public function getSyncedCreationAttributes(): array
+    {
+        // Default values when creating resources from tenant to central DB
+        return
+            [
+                'name' => 'Default Name',
+                'email' => 'default@localhost',
+                'password' => 'password',
+                'role' => 'admin',
+                'foo' => 'bar'
+            ];
+    }
+}
+
+// override method in ResourceUser class to return attribute names
+class ResourceUserProvidingAttributeNames extends ResourceUser
+{
+    public function getSyncedCreationAttributes(): array
+    {
+        // Attributes used when creating resources from tenant to central DB
+        // Notice here we are not adding "code" filed because it doesn't
+        // exist in central model
+        return
+            [
+                'name',
+                'password',
+                'email',
+                'role',
+                'foo' => 'bar'
+            ];
+    }
+
+}
+
+// override method in CentralUser class to return attribute default values
+class CentralUserProvidingDefaultValues extends CentralUser
```

Comment:
@abrardev99 I will merge this PR now so that the related tasks can be worked on tomorrow, but there are still some minor things you should fix tomorrow.

Re-read what I wrote on BC about the `ResourceUser` terminology: https://3.basecamp.com/5170965/buckets/23651018/messages/5438582828#__recording_5498105640

If e.g. `ResourceUserProvidingDefaultValues` is meant to be the counterpart/opposite to `CentralUserProvidingDefaultValues`, then wouldn't it make more sense to use `TenantUserProvidingDefaultValues`?

Reply:
`ResourceUser` exists from previous versions of this file, but I think back then there were only two main classes:
- Just a "User" class (named `ResourceUser` to avoid conflicts, as I explained)
- `CentralUser` class that specifically says that it's central

But now that we have many more models, and they're always in pairs (`XUserProviding...`), I'm thinking that it'd make more sense to name them `TenantUser*`.

That said, **I'd only do this for the suffixed classes**, since we don't want to use a generic name like `TenantUser` in this test. In that **one** case, the `Resource` prefix does make sense.

So we want to:
- Use `Tenant` instead of `Resource` in all suffixed (e.g. `X Providing ...`) models
- Either keep `ResourceUser` as it is, or rename it to `ResourceTenantUser`. Or maybe just add a comment saying `// Tenant users`

Reply:
@stancl Sent PR https://github.com/archtechx/tenancy/pull/992 for improving names.